### PR TITLE
Refactor test_common transforms section, introduce Hypothesis

### PR DESCRIPTION
**Pull request by JevinJ**
### Overview

This breaks up transforms related tests in test_common.py, specifically Common.transform(), rotate(), translate(). They are now separate tests, grouped and renamed to what is expected to happen. This also introduces [Hypothesis](https://hypothesis.readthedocs.io/en/latest/) to the testing requirements.

There are now failed tests which reveal possible "issues" that I'd like to discuss.

- If Common.transform is given values of nan or inf it will not have the same result as vtk

- Same for rotate

- Common.translate cannot handle addition of very large integers, likely overflow, specifically for example (+/-)9223372036854775809, which is just over size of long long.

### CODE REVIEW

**Comment by akaszynski** [comment]
These tests are very human readable, and I can see why you want to use hypothesis.

**Comment by JevinJ** [comment]
Hypothesis is pretty cool, it's easy to get results quickly. Just put declare kind of input you want and it gives it to you, it's also easy to create your own strategies by building on what is offered. It can be slow as each test is given about 200 runs by default, but that can be configured as well.

**Comment by JevinJ** [comment]
There are a couple options to fix the errors.

Convert nan/inf to 0 and large numbers(what number it should be depends on what vtk can handle, which I don't know). 

Or just fail if given that kind of input.

File: tests/test_common.py [hunk 1]
```diff
@@ -113,90 +125,61 @@ def test_copy(grid):
...
-    trans_filter = vtk.vtkTransformFilter()
-    trans_filter.SetTransform(trans)
-    trans_filter.SetInputData(grid)
-    trans_filter.Update()
-    grid_a = pyvista.UnstructuredGrid(trans_filter.GetOutput())
-
-    grid_b = grid.copy()
-    grid_b.rotate_z(angle)
-    assert np.allclose(grid_a.points, grid_b.points)
+class TestTransform:
```
Comment (imsodin) [inline tests/test_common.py:128]: Purely personal preference (i.e. feel free to ignore):  
I prefer flat over nested whenever possible. In this case I'd prefer grouping in files or no grouping at all over classes (assuming the class has no other purpose than grouping).

Comment (akaszynski) [reply tests/test_common.py:128]: I agree with this.  Classes also make it a bit harder to debug as well.  We can still group this using:
```python
###############################################################################
# Transform
###############################################################################
```

Comment (JevinJ) [reply tests/test_common.py:128]: More specifically the error is not with vtk but with numpy:

    TypeError: ufunc 'add' output (typecode 'O') could not be coerced to provided output parameter (typecode 'd') according to the casting rule ''same_kind''

Comment (JevinJ) [reply tests/test_common.py:128]: And on classes, it makes sense to me to have not only grouping by file, but by the specific method being tested. Why specifically would it be harder to debug if they're grouped in this way? I can see it could be harder to read on a console/terminal, pycharm gives a nice tree structure when grouped like this so I'm probably biased. But if multiple people agree it's easier to consider not doing so.

Comment (imsodin) [reply tests/test_common.py:128]: I almost feel somewhat bad having this brought up, this should definitely not block this change. It's purely style and while consistent coding style is good, there's neither a precedent for this in pyvista nor frankly is styling in general very consistent (or at least not enforced to be). And I don't get the difficulty regarding debugging either.

Comment (akaszynski) [reply tests/test_common.py:128]: The debugging behavior is a bit out of laziness on my end.  I'm fine with grouping by classes, and I agree that it makes the code easier to follow and to know where to add additional tests.

Comment (banesullivan) [reply tests/test_common.py:128]: FWIW, I don't like grouping by classes in tests - it's unnecessary.

Comment (JevinJ) [reply tests/test_common.py:128]: @imsodin 

No worries, it's a big style change so it's good to have input/consensus on this. I'm just not sure how I want to handle the other issues.

Comment (akaszynski) [reply tests/test_common.py:128]: Other issues can be fixed by not mitigating the failures that occur on VTK's side just yet, though it's helpful to know that they occur and we might consider mitigating it in the future on our side.  I agree with @imsodin about garbage in/garbage out and we should implement `equal_nan=True`.

**Comment by JevinJ** [comment]
This is ready for review/comments.

**Comment by akaszynski** [comment]
> This is ready for further review/comments. The remaining failing CI tests are due to hypothesis missing, which I can't fix.

This just needs a merge with master and adding `hypothesis` to `environment.yml`.

> I'm trying not to change any behavior about the tests, only move what is already there and add more tests if I can. There are a few cases where I shortened things, like there doesn't need to be a test for each axis when the setup is the same, I just parameterized and used getattr to make all 3 use the same test.

I see this, and here I think is a good use case for `hypothesis` as we don't have to hardcode the values.  Parameterizing `RotateX`... also simplifies the tests at the expense of readability, though I think it's warranted here as there's now only one test to maintain.

**Comment by akaszynski** [comment]
If I don't hear back from any other reviewers, I'll merge this tomorrow (Friday).

File: tests/test_common.py [hunk 2]
```diff
@@ -127,65 +138,36 @@ def test_transform(grid):
     grid_a.transform(trans)
     grid_b.transform(trans.GetMatrix())
     grid_c.transform(pyvista.trans_from_matrix(trans.GetMatrix()))
-    assert np.allclose(grid_a.points, grid_b.points)
-    assert np.allclose(grid_a.points, grid_c.points)
+    assert np.allclose(grid_a.points, grid_b.points, equal_nan=True)
+    assert np.allclose(grid_a.points, grid_c.points, equal_nan=True)
```
Comment (banesullivan) [inline tests/test_common.py:142]: Hopefully, there are no NaN values in the points?...

Comment (JevinJ) [reply tests/test_common.py:142]: This was discussed in the issue above by @imsodin.

Hypothesis just generates a range of inputs of the types and rules you specify, including random and common edge cases.
